src/Importify/Cabal.hs

```diff
@@ -3,28 +3,102 @@
 
 module Importify.Cabal
        ( getLibs
+       , getExtensionMaps
        , readCabal
+       , moduleNameToPath
+       , TargetMap
+       , ExtensionsMap
        ) where
 
-import           Universum
+import           Universum                             hiding (fromString)
 
+import           Data.List                             (nub)
+import qualified Data.Map.Strict                       as Map
+import           Distribution.ModuleName               (ModuleName, fromString,
+                                                        toFilePath)
 import           Distribution.Package                  (Dependency (..), PackageName (..))
-import           Distribution.PackageDescription       (GenericPackageDescription (..),
-                                                        condTreeData, libBuildInfo,
-                                                        targetBuildDepends)
+import           Distribution.PackageDescription       (BuildInfo, Executable,
+                                                        GenericPackageDescription (..),
+                                                        Library, buildInfo,
+                                                        condExecutables, condTreeData,
+                                                        defaultExtensions, exeModules,
+                                                        libBuildInfo, libModules,
+                                                        modulePath, targetBuildDepends)
 import           Distribution.PackageDescription.Parse (readPackageDescription)
 import           Distribution.Verbosity                (normal)
+import           Language.Haskell.Extension            (Extension (..))
+
+type TargetMap = Map String String
```

Comment:
Yes, exactly. You can find performance comparison in this repo: 

https://github.com/haskell-perf/dictionaries